src/System.Private.CoreLib/src/System/Reflection/Assembly.CoreCLR.cs

```diff
@@ -15,13 +15,19 @@ namespace System.Reflection
 {
     public abstract partial class Assembly : ICustomAttributeProvider, ISerializable
     {
-        // Locate an assembly by the long form of the assembly name. 
+        // Locate an assembly by the long form of the assembly name.
         // eg. "Toolbox.dll, version=1.1.10.1220, locale=en, publickey=1234567890123456789012345678901234567890"
         [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
         public static Assembly Load(string assemblyString)
         {
+            IntPtr nativeAssemblyLoadContext = IntPtr.Zero;
+            if (AssemblyLoadContext.CurrentContextualReflectionContext != null)
```

Comment:
The argument passed to VM may better me the managed pointer. It will guarantee that the context won't get unloaded while being in use.

Reply:
The thread is holding it in its copy of the `CurrentContextualReflectionContext`.  It shouldn't be unloaded.

Reply:
The reference to the ALC in `CurrentContextualReflectionContext` is not guaranteed to stay there. It can get overwritten.

I am not sure whether it is possible via valid usage patterns, but it is certainly possible to do if one uses the API incorrectly. We should not end up with dangling pointers if the managed API is used incorrectly.

Reply:
I pushed a patch to pass AssemblyLoadContext as a managed object for these cases